### ISSUE TITLE
[BUG] TypeError: unsupported operand type(s) for /: 'str' and 'int' when using remove_multicollinearity = True in clustering setup

### ISSUE THREAD

**Issue by tdekelver** [pycaret/pycaret#1325]
**Describe the bug**
Whenever I include the `remove_multicollinearity = True` parameter to the `setup` of clustering, an error is thrown just after confirming the inferred data types. If I put the parameter to False, everything works as expected.

**To Reproduce**

```python
from pycaret.datasets import get_data
dataset = get_data('mice')

exp_clu101 = setup(dataset, normalize = True, 
                   ignore_features = ['MouseID'],
                   session_id = 123, remove_multicollinearity = True)
```

**Traceback**
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-26-100eea955615> in <module>
      1 exp_clu101 = setup(dataset, normalize = True, 
      2                    ignore_features = ['MouseID'],
----> 3                    session_id = 123, remove_multicollinearity = True)

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\pycaret\clustering.py in setup(data, preprocess, imputation_type, iterative_imputation_iters, categorical_features, categorical_imputation, categorical_iterative_imputer, ordinal_features, high_cardinality_features, high_cardinality_method, numeric_features, numeric_imputation, numeric_iterative_imputer, date_features, ignore_features, normalize, normalize_method, transformation, transformation_method, handle_unknown_categorical, unknown_categorical_method, pca, pca_method, pca_components, ignore_low_variance, combine_rare_levels, rare_level_threshold, bin_numeric_features, remove_multicollinearity, multicollinearity_threshold, remove_perfect_collinearity, group_features, group_names, n_jobs, use_gpu, custom_pipeline, html, session_id, log_experiment, experiment_name, log_plots, log_profile, log_data, silent, verbose, profile, profile_kwargs)
    442         verbose=verbose,
    443         profile=profile,
--> 444         profile_kwargs=profile_kwargs,
    445     )
    446 

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\pycaret\internal\tabular.py in setup(data, target, ml_usecase, available_plots, train_size, test_data, preprocess, imputation_type, iterative_imputation_iters, categorical_features, categorical_imputation, categorical_iterative_imputer, ordinal_features, high_cardinality_features, high_cardinality_method, numeric_features, numeric_imputation, numeric_iterative_imputer, date_features, ignore_features, normalize, normalize_method, transformation, transformation_method, handle_unknown_categorical, unknown_categorical_method, pca, pca_method, pca_components, ignore_low_variance, combine_rare_levels, rare_level_threshold, bin_numeric_features, remove_outliers, outliers_threshold, remove_multicollinearity, multicollinearity_threshold, remove_perfect_collinearity, create_clusters, cluster_iter, polynomial_features, polynomial_degree, trigonometry_features, polynomial_threshold, group_features, group_names, feature_selection, feature_selection_threshold, feature_selection_method, feature_interaction, feature_ratio, interaction_threshold, fix_imbalance, fix_imbalance_method, transform_target, transform_target_method, data_split_shuffle, data_split_stratify, fold_strategy, fold, fold_shuffle, fold_groups, n_jobs, use_gpu, custom_pipeline, html, session_id, log_experiment, experiment_name, log_plots, log_profile, log_data, silent, verbose, profile, profile_kwargs, display)
   1350         y = data[target]
   1351     else:
-> 1352         X = prep_pipe.fit_transform(train_data).drop(target, axis=1)
   1353         X_train = X
   1354 

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\sklearn\pipeline.py in fit_transform(self, X, y, **fit_params)
    365         """
    366         fit_params_steps = self._check_fit_params(**fit_params)
--> 367         Xt = self._fit(X, y, **fit_params_steps)
    368 
    369         last_step = self._final_estimator

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\sklearn\pipeline.py in _fit(self, X, y, **fit_params_steps)
    294                 message_clsname='Pipeline',
    295                 message=self._log_message(step_idx),
--> 296                 **fit_params_steps[name])
    297             # Replace the transformer of the step with the fitted
    298             # transformer. This is necessary when loading the transformer

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\joblib\memory.py in __call__(self, *args, **kwargs)
    350 
    351     def __call__(self, *args, **kwargs):
--> 352         return self.func(*args, **kwargs)
    353 
    354     def call_and_shelve(self, *args, **kwargs):

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\sklearn\pipeline.py in _fit_transform_one(transformer, X, y, weight, message_clsname, message, **fit_params)
    738     with _print_elapsed_time(message_clsname, message):
    739         if hasattr(transformer, 'fit_transform'):
--> 740             res = transformer.fit_transform(X, y, **fit_params)
    741         else:
    742             res = transformer.fit(X, y, **fit_params).transform(X)

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\pycaret\internal\preprocess.py in fit_transform(self, data, y)
   2695             data frame
   2696     """
-> 2697         self.fit(data)
   2698         return self.transform(data)
   2699 

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\pycaret\internal\preprocess.py in fit(self, data, y)
   2507         # self.data_c = self.data1.T.drop_duplicates()
   2508         # self.data1 = self.data_c.T
-> 2509         corr = pd.DataFrame(np.corrcoef(self.data1.T))
   2510         corr.columns = self.data1.columns
   2511         corr.index = self.data1.columns

<__array_function__ internals> in corrcoef(*args, **kwargs)

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\numpy\lib\function_base.py in corrcoef(x, y, rowvar, bias, ddof)
   2549         warnings.warn('bias and ddof have no effect and are deprecated',
   2550                       DeprecationWarning, stacklevel=3)
-> 2551     c = cov(x, y, rowvar)
   2552     try:
   2553         d = diag(c)

<__array_function__ internals> in cov(*args, **kwargs)

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\numpy\lib\function_base.py in cov(m, y, rowvar, bias, ddof, fweights, aweights)
   2454             w *= aweights
   2455 
-> 2456     avg, w_sum = average(X, axis=1, weights=w, returned=True)
   2457     w_sum = w_sum[0]
   2458 

<__array_function__ internals> in average(*args, **kwargs)

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\numpy\lib\function_base.py in average(a, axis, weights, returned)
    378 
    379     if weights is None:
--> 380         avg = a.mean(axis)
    381         scl = avg.dtype.type(a.size/avg.size)
    382     else:

C:\ProgramData\Internal\Anaconda3\envs\SAM3.7\lib\site-packages\numpy\core\_methods.py in _mean(a, axis, dtype, out, keepdims)
    161     if isinstance(ret, mu.ndarray):
    162         ret = um.true_divide(
--> 163                 ret, rcount, out=ret, casting='unsafe', subok=False)
    164         if is_float16_result and out is None:
    165             ret = arr.dtype.type(ret)

TypeError: unsupported operand type(s) for /: 'str' and 'int'
```

**Versions**
<details>
pycaret==2.3.0
</details>

**Comment by Yard1** [comment]
Can you update to 2.3.1?

**Comment by pycaret** [comment]
@tdekelver Hi.. Can you confirm if the issue is fixed after the pycaret update to 2.3.1?

Thanks for your help.

**Comment by wkuopt** [comment]
Hi, an upgrade should fix this issue. I was testing with version 2.3.4 and the setup was done successfully.
Code for testing:
```
def test_clustering_operand_error():
    dataset = pycaret.datasets.get_data('mice')

    exp_clu101 = pycaret.clustering.setup(dataset,
                                          ignore_features=['MouseID'],
                                          remove_multicollinearity=True,
                                          normalize=True,
                                          log_experiment=True,
                                          silent=True,
                                          html=False,
                                          session_id=123,
                                          n_jobs=1,
)
```
Log:
```
2022-01-08 14:18:51,666:INFO:PyCaret Supervised Module
2022-01-08 14:18:51,666:INFO:ML Usecase: clustering
2022-01-08 14:18:51,667:INFO:version 2.3.4
2022-01-08 14:18:51,667:INFO:Initializing setup()
2022-01-08 14:18:51,667:INFO:setup(display=None, profile_kwargs=None, profile=False, verbose=True, silent=True, log_data=False, log_profile=False, log_plots=False, experiment_name=None, log_experiment=True, session_id=123, html=False, custom_pipeline=None, use_gpu=False, n_jobs=1, fold_groups=None, fold_shuffle=False, fold=10, fold_strategy=kfold, data_split_stratify=False, data_split_shuffle=False, transform_target_method=box-cox, transform_target=False, fix_imbalance_method=None, fix_imbalance=False, interaction_threshold=0.01, feature_ratio=False, feature_interaction=False, feature_selection_method=classic, feature_selection_threshold=0.8, feature_selection=False, group_names=None, group_features=None, polynomial_threshold=0.1, trigonometry_features=False, polynomial_degree=2, polynomial_features=False, cluster_iter=20, create_clusters=False, remove_perfect_collinearity=False, multicollinearity_threshold=0.9, remove_multicollinearity=True, outliers_threshold=0.05, remove_outliers=False, bin_numeric_features=None, rare_level_threshold=0.1, combine_rare_levels=False, ignore_low_variance=False, pca_components=None, pca_method=linear, pca=False, unknown_categorical_method=least_frequent, handle_unknown_categorical=True, transformation_method=yeo-johnson, transformation=False, normalize_method=zscore, normalize=True, ignore_features=['MouseID'], date_features=None, numeric_iterative_imputer=lightgbm, numeric_imputation=mean, numeric_features=None, high_cardinality_method=frequency, high_cardinality_features=None, ordinal_features=None, categorical_iterative_imputer=lightgbm, categorical_imputation=mode, categorical_features=None, iterative_imputation_iters=5, imputation_type=simple, preprocess=True, test_data=None, train_size=0.7, available_plots={'cluster': 'Cluster PCA Plot (2d)', 'tsne': 'Cluster TSnE (3d)', 'elbow': 'Elbow', 'silhouette': 'Silhouette', 'distance': 'Distance', 'distribution': 'Distribution'}, ml_usecase=clustering, target=None)
2022-01-08 14:18:51,667:INFO:Checking environment
2022-01-08 14:18:51,667:INFO:python_version: 3.6.12
2022-01-08 14:18:51,667:INFO:python_build: ('default', 'Sep  9 2020 00:29:25')
2022-01-08 14:18:51,667:INFO:machine: AMD64
2022-01-08 14:18:51,668:INFO:platform: Windows-10-10.0.18362-SP0
2022-01-08 14:18:51,668:INFO:Memory: svmem(total=16981368832, available=1788686336, percent=89.5, used=15192682496, free=1788686336)
2022-01-08 14:18:51,668:INFO:Physical Core: 4
2022-01-08 14:18:51,668:INFO:Logical Core: 8
2022-01-08 14:18:51,668:INFO:Checking libraries
2022-01-08 14:18:51,668:INFO:pd==1.1.4
2022-01-08 14:18:51,668:INFO:numpy==1.19.4
2022-01-08 14:18:51,668:INFO:sklearn==0.23.2
2022-01-08 14:18:51,668:INFO:lightgbm==3.1.0
2022-01-08 14:18:51,686:INFO:catboost==0.24.3
2022-01-08 14:18:51,713:INFO:xgboost==1.2.1
2022-01-08 14:18:52,911:INFO:mlflow==1.12.1
2022-01-08 14:18:52,913:INFO:Checking Exceptions
2022-01-08 14:18:52,913:INFO:Declaring global variables
2022-01-08 14:18:52,913:INFO:USI: 1783
2022-01-08 14:18:52,914:INFO:pycaret_globals: {'_gpu_n_jobs_param', 'display_container', 'fold_shuffle_param', 'transform_target_method_param', 'imputation_regressor', 'fold_param', 'html_param', 'fold_groups_param', 'target_param', 'X', 'iterative_imputation_iters_param', 'n_jobs_param', 'master_model_container', 'imputation_classifier', 'create_model_container', 'gpu_param', 'stratify_param', 'logging_param', 'log_plots_param', 'exp_name_log', '_all_models_internal', '_available_plots', 'pycaret_globals', '_internal_pipeline', '_all_metrics', 'fix_imbalance_param', 'fold_generator', 'transform_target_param', 'fold_groups_param_full', 'USI', 'data_before_preprocess', '_all_models', 'fix_imbalance_method_param', 'experiment__', 'prep_pipe', '_ml_usecase', 'seed'}
2022-01-08 14:18:52,914:INFO:Preparing display monitor
2022-01-08 14:18:52,914:INFO:Preparing display monitor
2022-01-08 14:18:52,914:INFO:Importing libraries
2022-01-08 14:18:52,914:INFO:Copying data for preprocessing
2022-01-08 14:18:52,915:INFO:Declaring preprocessing parameters
2022-01-08 14:18:52,926:INFO:Creating preprocessing pipeline
2022-01-08 14:18:52,955:INFO:Preprocessing pipeline created successfully
2022-01-08 14:18:52,955:ERROR:(Process Exit): setup has been interupted with user command 'quit'. setup must rerun.
2022-01-08 14:18:52,955:INFO:Creating global containers
2022-01-08 14:18:52,957:INFO:Internal pipeline: Pipeline(memory=None, steps=[('empty_step', 'passthrough')], verbose=False)
2022-01-08 14:18:53,276:INFO:Creating grid variables
2022-01-08 14:18:53,288:INFO:Logging experiment in MLFlow
2022-01-08 14:18:53,951:INFO:SubProcess save_model() called ==================================
2022-01-08 14:18:53,974:INFO:Initializing save_model()
2022-01-08 14:18:53,974:INFO:save_model(kwargs={}, verbose=False, prep_pipe_=Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=False,
                                      features_todrop=['MouseID'],
                                      id_columns=[], ml_usecase='regression',
                                      numerical_features=[],
                                      target='UNSUPERVISED_DUMMY_TARGET',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='most frequent',
                                fill_value_categorical=None,
                                fill_value_nume...
                ('dummy', Dummify(target='UNSUPERVISED_DUMMY_TARGET')),
                ('fix_perfect', 'passthrough'),
                ('clean_names', Clean_Colum_Names()),
                ('feature_select', 'passthrough'),
                ('fix_multi',
                 Fix_multicollinearity(correlation_with_target_preference=None,
                                       correlation_with_target_threshold=0.0,
                                       target_variable='UNSUPERVISED_DUMMY_TARGET',
                                       threshold=0.9)),
                ('dfs', 'passthrough'), ('pca', 'passthrough')],
         verbose=False), model_name=Transformation Pipeline, model=Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=False,
                                      features_todrop=['MouseID'],
                                      id_columns=[], ml_usecase='regression',
                                      numerical_features=[],
                                      target='UNSUPERVISED_DUMMY_TARGET',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='most frequent',
                                fill_value_categorical=None,
                                fill_value_nume...
                ('dummy', Dummify(target='UNSUPERVISED_DUMMY_TARGET')),
                ('fix_perfect', 'passthrough'),
                ('clean_names', Clean_Colum_Names()),
                ('feature_select', 'passthrough'),
                ('fix_multi',
                 Fix_multicollinearity(correlation_with_target_preference=None,
                                       correlation_with_target_threshold=0.0,
                                       target_variable='UNSUPERVISED_DUMMY_TARGET',
                                       threshold=0.9)),
                ('dfs', 'passthrough'), ('pca', 'passthrough')],
         verbose=False))
2022-01-08 14:18:53,974:INFO:Adding model into prep_pipe
2022-01-08 14:18:53,976:WARNING:Only Model saved as it was a pipeline.
2022-01-08 14:18:53,986:INFO:Transformation Pipeline.pkl saved in current working directory
2022-01-08 14:18:53,993:INFO:Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=False,
                                      features_todrop=['MouseID'],
                                      id_columns=[], ml_usecase='regression',
                                      numerical_features=[],
                                      target='UNSUPERVISED_DUMMY_TARGET',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='most frequent',
                                fill_value_categorical=None,
                                fill_value_nume...
                ('dummy', Dummify(target='UNSUPERVISED_DUMMY_TARGET')),
                ('fix_perfect', 'passthrough'),
                ('clean_names', Clean_Colum_Names()),
                ('feature_select', 'passthrough'),
                ('fix_multi',
                 Fix_multicollinearity(correlation_with_target_preference=None,
                                       correlation_with_target_threshold=0.0,
                                       target_variable='UNSUPERVISED_DUMMY_TARGET',
                                       threshold=0.9)),
                ('dfs', 'passthrough'), ('pca', 'passthrough')],
         verbose=False)
2022-01-08 14:18:53,993:INFO:save_model() successfully completed......................................
2022-01-08 14:18:54,074:INFO:SubProcess save_model() end ==================================
2022-01-08 14:18:54,138:INFO:create_model_container: 0
2022-01-08 14:18:54,138:INFO:master_model_container: 0
2022-01-08 14:18:54,138:INFO:display_container: 1
2022-01-08 14:18:54,147:INFO:Pipeline(memory=None,
         steps=[('dtypes',
                 DataTypes_Auto_infer(categorical_features=[],
                                      display_types=False,
                                      features_todrop=['MouseID'],
                                      id_columns=[], ml_usecase='regression',
                                      numerical_features=[],
                                      target='UNSUPERVISED_DUMMY_TARGET',
                                      time_features=[])),
                ('imputer',
                 Simple_Imputer(categorical_strategy='most frequent',
                                fill_value_categorical=None,
                                fill_value_nume...
                ('dummy', Dummify(target='UNSUPERVISED_DUMMY_TARGET')),
                ('fix_perfect', 'passthrough'),
                ('clean_names', Clean_Colum_Names()),
                ('feature_select', 'passthrough'),
                ('fix_multi',
                 Fix_multicollinearity(correlation_with_target_preference=None,
                                       correlation_with_target_threshold=0.0,
                                       target_variable='UNSUPERVISED_DUMMY_TARGET',
                                       threshold=0.9)),
                ('dfs', 'passthrough'), ('pca', 'passthrough')],
         verbose=False)
2022-01-08 14:18:54,148:INFO:setup() succesfully completed......................................
'''
